# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/preload/PreloadHashKey.h
# Commit: 7d5683180ac3
# Full Hash: 7d5683180ac3f9e7cc96d68dde27b8b48642a7ac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r a48aff5f2555 -r 7d5683180ac3 uriloader/preload/PreloadHashKey.h
--- a/uriloader/preload/PreloadHashKey.h	Thu Feb 17 15:57:52 2022 +0000
+++ b/uriloader/preload/PreloadHashKey.h	Thu Feb 17 15:57:52 2022 +0000
@@ -7,11 +7,15 @@
 
 #include "mozilla/CORSMode.h"
 #include "mozilla/css/SheetParsingMode.h"
-#include "mozilla/dom/ScriptKind.h"
+#include "js/loader/ScriptKind.h"
 #include "nsURIHashKey.h"
 
 class nsIPrincipal;
 
+namespace JS::loader {
+enum class ScriptKind;
+}
+
 namespace mozilla {
 
 namespace css {
@@ -44,7 +48,7 @@
 
   // Construct key for "script"
   static PreloadHashKey CreateAsScript(nsIURI* aURI, CORSMode aCORSMode,
-                                       dom::ScriptKind aScriptKind);
+                                       JS::loader::ScriptKind aScriptKind);
   static PreloadHashKey CreateAsScript(nsIURI* aURI,
                                        const nsAString& aCrossOrigin,
                                        const nsAString& aType);
@@ -92,7 +96,7 @@
   nsCOMPtr<nsIPrincipal> mPrincipal;
 
   struct {
-    dom::ScriptKind mScriptKind = dom::ScriptKind::eClassic;
+    JS::loader::ScriptKind mScriptKind = JS::loader::ScriptKind::eClassic;
   } mScript;
 
   struct {
